src/RX.cpp

```diff
@@ -135,19 +123,17 @@ bool RX::match (
   if (! _compiled)
     compile ();
 
-  regmatch_t rm[2];
-  int offset = 0;
-  int length = in.length ();
-  while (regexec (&_regex, in.c_str () + offset, 2, &rm[0], 0) == 0 &&
-         offset < length)
-  {
-    start.push_back (rm[0].rm_so + offset);
-    end.push_back   (rm[0].rm_eo + offset);
-    offset += rm[0].rm_eo;
+  std::smatch sm;
+  std::string::const_iterator search_start(in.cbegin());
+  std::string::const_iterator search_end(in.cend());
+
+  while (std::regex_search(search_start, search_end, sm, _regex)) {
```

Comment:
Similarly here!

Reply:
Where exactly can I simplify? With an additional check that `search_start` does not go beyond the end of `in`, both variables are used.

Reply:
Oh, I missed that search_start was used below in this loop. Sorry!